packages/server/src/open-feature.ts

```diff
@@ -71,22 +73,107 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI {
+    }
+
+    if (client) {
+      this._providers.set(client, provider);
+    } else {
       this._provider = provider;
     }
+
     return this;
   }
 
-  getClient(name?: string, version?: string, context?: EvaluationContext): Client {
+  /**
+   * A factory function for creating new unnamed OpenFeature clients. Clients can contain
+   * their own state (e.g. logger, hook, context). Multiple clients can be used
+   * to segment feature flag configuration.
+   *
+   * All unnamed clients use the same provider set via {@link this.setProvider setProvider}.
+   *
+   * @param {string} name The name of the client
+   * @param {string} version The version of the client (only used for metadata)
+   * @param {EvaluationContext} context Evaluation context that should be set on the client to used during flag evaluations
+   * @returns {Client} OpenFeature Client
+   */
+  getClient(context?: EvaluationContext): Client;
+  /**
+   * A factory function for creating new named OpenFeature clients. Clients can contain
+   * their own state (e.g. logger, hook, context). Multiple clients can be used
+   * to segment feature flag configuration.
+   *
+   * If there is already a provider bound to this name via {@link this.setProvider setProvider}, this provider will be used.
+   * Otherwise, the default provider is used until a provider is assigned to that name.
+   *
+   * @param {string} name The name of the client
+   * @param {EvaluationContext} context Evaluation context that should be set on the client to used during flag evaluations
+   * @returns {Client} OpenFeature Client
+   */
+  getClient(name: string, context?: EvaluationContext): Client;
+  /**
+   * A factory function for creating new named OpenFeature clients. Clients can contain
+   * their own state (e.g. logger, hook, context). Multiple clients can be used
+   * to segment feature flag configuration.
+   *
+   * If there is already a provider bound to this name via {@link this.setProvider setProvider}, this provider will be used.
+   * Otherwise, the default provider is used until a provider is assigned to that name.
+   *
+   * @param {string} name The name of the client
+   * @param {string} version The version of the client (only used for metadata)
+   * @param {EvaluationContext} context Evaluation context that should be set on the client to used during flag evaluations
+   * @returns {Client} OpenFeature Client
+   */
+  getClient(name: string, version: string, context?: EvaluationContext): Client;
+  getClient(
+    nameOrContext?: string | EvaluationContext,
+    versionOrContext?: string | EvaluationContext,
+    contextOrUndefined?: EvaluationContext
+  ): Client {
+    const name = stringOrUndefined(nameOrContext);
+    const version = stringOrUndefined(versionOrContext);
+    const context =
+      objectOrUndefined<EvaluationContext>(nameOrContext) ??
+      objectOrUndefined<EvaluationContext>(versionOrContext) ??
+      objectOrUndefined<EvaluationContext>(contextOrUndefined);
```

Comment:
Nice 😄

Reply:
What does that mean? 😅
Should I change that?

Reply:
No, all good. It's a nice way to handle relatively complex overloads.